packages/database/src/migrations/20210419225536-UpdatePSSSTotalSitesReportToUseSiteData-modifies-data.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+import { generateId, insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const oldConfig = {
+  formula: 'PSSS_Sites_Reported',
+  aggregation: 'FINAL_EACH_WEEK',
+};
+
+const newConfig = {
+  formula: `firstExistingValue(PSSS_Sites_Reported, PSSS_Site_Weekly_Survey_Completed, PSSS_Site_Daily_Survey_Completed)`,
+  aggregation: {
+    PSSS_Sites_Reported: 'FINAL_EACH_WEEK',
+    PSSS_Site_Weekly_Survey_Completed: [
+      'FINAL_EACH_WEEK',
+      {
+        type: 'SUM_PER_PERIOD_PER_ORG_GROUP',
+        config: { dataSourceEntityType: 'facility', aggregationEntityType: 'country' },
+      },
+    ],
+    PSSS_Site_Daily_Survey_Completed: [
+      'FINAL_EACH_WEEK',
+      {
+        type: 'SUM_PER_PERIOD_PER_ORG_GROUP',
+        config: { dataSourceEntityType: 'facility', aggregationEntityType: 'country' },
+      },
+    ],
+  },
+  defaultValues: {
+    PSSS_Sites_Reported: 'undefined',
+    PSSS_Site_Weekly_Survey_Completed: 'undefined',
+    PSSS_Site_Daily_Survey_Completed: 'undefined',
+  },
+};
+
+const PSSS_SITE_REPORTED_WEEKLY = {
+  code: 'PSSS_Site_Weekly_Survey_Completed',
+  builder: 'eventCheckConditions',
+  config: {
+    formula: 'true',
+    programCode: 'PSSS_WSR',
+  },
+};
```

Comment:
Technically these indicators could be parameters, however there's currently a bug in indicator parameters that the fetch options don't get mutated as required per the aggregation options. I wasn't entirely sure if I ought to go ahead and make that change, since it more tightly couples the aggregator and indicator logic and seemed pretty messy... I think we can come up with a better design with time.

For now this seems acceptable to me 😄

Reply:
Good and pragmatic approach, I agree!